### PR TITLE
update fishing

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
adds fishing focuses - ore, trash, fishing; find them in the misc crafting menu
doubles the chance (not much) of getting strange rocks from fishing
fixes the water basin so it can actually be upgraded by a journeyman smithy

## How This Contributes To The Skyrat Roleplay Experience
fishing more "enjoyable," and fixing an issue with the water basin.

## Changelog

:cl:
add: added fishing attachments (check misc crafting tab)
balance: doubles strange rock chance in fishing
fix: water basins can be upgraded now by journeyman smithies
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 1]
```diff
@@ -71,28 +77,53 @@ GLOBAL_LIST_INIT(fishing_weights, list(
 		QDEL_NULL(mutate_parent)
 	if(!fisher)
 		return
+	var/obj/item/fishing_rod/fisher_rod = fisher
```
Comment (Zonespace27) [inline modular_skyrat/modules/fishing/code/fishing.dm]: couldn't we just typecast `fisher` to be `obj/item/fishing_rod` in the `finish_fishing()` proc? I don't think it'd be anything else

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 2]
```diff
@@ -187,6 +236,48 @@ GLOBAL_LIST_INIT(fishing_weights, list(
 		RegisterSignal(target_atom, COMSIG_MOVABLE_MOVED, .proc/check_movement, override = TRUE)
 	SEND_SIGNAL(target_atom, COMSIG_START_FISHING)
 
+/obj/item/fishing_rod/attackby(obj/item/I, mob/living/user, params)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fishing/code/fishing.dm]: one letter var

**Comment by Stalkeros2** [comment]
No way, Burgerstation.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 3]
```diff
@@ -187,6 +235,48 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+/obj/item/fishing_rod/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(attacking_item.tool_behaviour == TOOL_CROWBAR)
+		var/obj/item/fishing_focus/find_focus = locate() in contents
+		if(find_focus)
+			find_focus.forceMove(get_turf(src))
+		fishing_focus = FOCUS_NONE
+		attacking_item.play_tool_sound(src, 50)
+		cut_overlays()
+		return
+	if(istype(attacking_item, /obj/item/fishing_focus))
```
Comment (Zonespace27) [inline modular_skyrat/modules/fishing/code/fishing.dm:247]:
```suggestion
	else if(istype(attacking_item, /obj/item/fishing_focus))
```

**Pull request merged**